python/pyarrow/types.py

```diff
@@ -59,92 +69,92 @@ def is_null(t):
 
 @doc(is_null, datatype="boolean")
 def is_boolean(t):
-    return t.id == lib.Type_BOOL
+    return _is_primitive(t) and _bit_width(t) == 1
```

Comment:
I mean remove `_is_primitive` and other functions, and implement `is_boolean` etc. in `types.pxi`.

Reply:
Okay... I will try that. Thank you for your response!